### ISSUE TITLE
Wrong "Expected one more positional argument" error when typing instance methods with positional arguments

### ISSUE THREAD

**Issue by danimelchor** [microsoft/pyright#9966]
**Describe the bug**
Using a protocol to type an instance method should allow the `self` parameter to be positional.

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

Fully working example

```python
import typing as t

T = t.TypeVar("T")


class WithFn(t.Protocol[T]):
    def __call__(self, __instance: T, __x: str, msg: str | None = None) -> T: ...


def decorator(fn: WithFn[T]) -> WithFn[T]:
    return fn


class Example:
    @decorator
    def foo(self, whatever: str, msg: str | None = None) -> t.Self:
        print(f"{whatever=} {msg=}")
        return self


e = Example()
e.foo("Some string :)")
e.foo("Some string :)", msg="Some message!")
```

Running it yields:
```
x='Some string :)' msg=None
x='Some string :)' msg='Some message!'
```

But pyright complains:
```
$ pyright tests.py
/.../
  /.../ - error: Expected 1 more positional argument (reportCallIssue)
  /.../tests.py:23:29 - error: Expected 1 more positional argument (reportCallIssue)
2 errors, 0 warnings, 0 informations
```

<img width="1133" alt="Image" src="https://github.com/user-attachments/assets/97166d4f-ece0-48cd-a737-3d80c8b5162f" />

**VS Code extension or command-line**
Running pyright both as an LSP and as a CLI on version `1.1.394`

**Comment by erictraut** [comment]
Pyright is working correctly here as per the Python typing spec, so I don't consider this a bug. Mypy also reports the same errors for your code sample.

The problem is that your decorator is transforming the method into a callable object whose signature (after binding the `self` parameter to an instance) contains two positional parameters (`__instance` and `__x`).

It's not clear to me from the sample what you're trying to do here, but if you'd like to provide more details, I might be able to provide advice about how to accomplish your goals using the facilities and behaviors provided in the Python type system. Or if you have general questions about the Python typing spec and how [callback protocols](https://typing.readthedocs.io/en/latest/spec/callables.html#callback-protocols) work, you could post your question to the [Python typing forum](https://discuss.python.org/c/typing/32).
